plugins/Transitions/API.php

```diff
@@ -656,4 +660,80 @@ protected function processTransitionsToFollowingPages($type, $actions)
             $this->totalTransitionsToFollowingPages += $actions;
         }
     }
+
+    /**
+     * Retrieve the period allowed config setting for a site or all sites if null
+     *
+     * @param $idSite
+     *
+     * @return string|null
+     */
+    public function getPeriodAllowedConfig($idSite) : ?string
+    {
+        $transitionsGeneralConfig = Config::getInstance()->Transitions;
+        $generalMaxPeriodAllowed = ($transitionsGeneralConfig && !empty($transitionsGeneralConfig['max_period_allowed']) ? $transitionsGeneralConfig['max_period_allowed']: null);
+
+        $siteMaxPeriodAllowed = null;
+        if ($idSite) {
+            $sectionName = 'Transitions_'.$idSite;
+            $transitionsSiteConfig = Config::getInstance()->$sectionName;
+            $siteMaxPeriodAllowed = ($transitionsSiteConfig && !empty($transitionsSiteConfig['max_period_allowed']) ? $transitionsSiteConfig['max_period_allowed'] : null);
+        }
+
+        if (!$generalMaxPeriodAllowed && !$siteMaxPeriodAllowed) {
+            return 'all'; // No config setting, so all periods are valid
+        }
+
+        // Site setting overrides general, if it exists
+        return ($siteMaxPeriodAllowed ? $siteMaxPeriodAllowed : $generalMaxPeriodAllowed);
```

Comment:
could be shortened to

```suggestion
        return $siteMaxPeriodAllowed ?? $generalMaxPeriodAllowed;
```